src/flb_output.c

```diff
@@ -807,277 +807,283 @@ const char *flb_output_get_property(const char *key, struct flb_output_instance
-        }
+    /* Override first configmap value based on it plugin flag */
+    m = mk_list_entry_first(ins->tls_config_map, struct flb_config_map, _head);
+    if (p->flags & FLB_IO_TLS) {
+        m->value.val.boolean = FLB_TRUE;
+    }
+    else {
+        m->value.val.boolean = FLB_FALSE;
+    }
 #endif
-        /*
-         * Validate 'net.*' properties: if the plugin use the Upstream interface,
-         * it might receive some networking settings.
-         */
-        if (mk_list_size(&ins->net_properties) > 0) {
-            ret = flb_config_map_properties_check(ins->p->name,
-                                                  &ins->net_properties,
-                                                  ins->net_config_map);
-            if (ret == -1) {
-                if (config->program_name) {
-                    flb_helper("try the command: %s -o %s -h\n",
-                               config->program_name, ins->p->name);
-                }
-                flb_output_instance_destroy(ins);
-                return -1;
-            }
-        }
-
-        /* Initialize plugin through it 'init callback' */
-        ret = p->cb_init(ins, config, ins->data);
+    /*
+        * Validate 'net.*' properties: if the plugin use the Upstream interface,
+        * it might receive some networking settings.
+        */
+    if (mk_list_size(&ins->net_properties) > 0) {
+        ret = flb_config_map_properties_check(ins->p->name,
+                                                &ins->net_properties,
+                                                ins->net_config_map);
         if (ret == -1) {
-            flb_error("[output] failed to initialize '%s' plugin",
-                      p->name);
+            if (config->program_name) {
+                flb_helper("try the command: %s -o %s -h\n",
+                            config->program_name, ins->p->name);
+            }
             flb_output_instance_destroy(ins);
             return -1;
         }
+    }
 
-        /* Multi-threading enabled if configured */
-        ret = flb_output_enable_multi_threading(ins, config);
-        if (ret == -1) {
-            flb_error("[output] could not start thread pool for '%s' plugin",
-                      flb_output_name(ins));
-            return -1;
-        }
+    /* Initialize plugin through it 'init callback' */
+    ret = p->cb_init(ins, config, ins->data);
+    if (ret == -1) {
+        flb_error("[output] failed to initialize '%s' plugin",
+                    p->name);
+        flb_output_instance_destroy(ins);
+        return -1;
+    }
+
+    /* Multi-threading enabled if configured */
+    ret = flb_output_enable_multi_threading(ins, config);
+    if (ret == -1) {
+        flb_error("[output] could not start thread pool for '%s' plugin",
+                    flb_output_name(ins));
+        return -1;
+    }
+}
+
+/* Trigger the output plugins setup callbacks to prepare them. */
+int flb_output_init_all(struct flb_config *config)
```

Comment:
The basic gist of this change is that I separated out the function of initializing output plugins into a separate function. This is necessary for connecting to output plugins from the trace code, at least as far as I can tell. I also felt it was appropriate to add this function there is already a counterpart on the input side.